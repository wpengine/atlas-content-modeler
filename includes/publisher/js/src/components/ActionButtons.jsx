import React from "react";
import { sprintf, __ } from "@wordpress/i18n";

export default function ActionButtons({ isEditMode }) {
	function clickHandler(e) {
		e.preventDefault();
		document.getElementById("publish").click();
	}

	return (
		<div className="flex-parent action-buttons">
			<div className="flex-grow"></div>
			<div>
				<button
					className="button button-primary button-large action-button"
					onClick={(e) => {
						clickHandler(e);
					}}
				>
					{isEditMode
						? __("Update", "atlas-content-modeler")
						: __("Publish", "atlas-content-modeler")}
				</button>
			</div>
		</div>
	);
}
