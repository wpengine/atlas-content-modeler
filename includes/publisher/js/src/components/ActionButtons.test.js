import React from "react";
import { act, create } from "react-test-renderer";
import ActionButtons from "./ActionButtons";

describe("ActionButtons", () => {
	let root;

	it("renders a publish button", () => {
		act(() => {
			root = create(<ActionButtons isEditMode={false} />);
		});

		expect(root.toJSON()).toMatchSnapshot();
	});

	it("renders an update button", () => {
		act(() => {
			root = create(<ActionButtons isEditMode={true} />);
		});

		expect(root.toJSON()).toMatchSnapshot();
	});
});
