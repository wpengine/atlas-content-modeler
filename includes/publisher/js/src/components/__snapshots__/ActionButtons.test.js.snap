// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ActionButtons renders a publish button 1`] = `
<div
  className="flex-parent action-buttons"
>
  <div
    className="flex-grow"
  />
  <div>
    <button
      className="button button-primary button-large action-button"
      onClick={[Function]}
    >
      Publish
    </button>
  </div>
</div>
`;

exports[`ActionButtons renders an update button 1`] = `
<div
  className="flex-parent action-buttons"
>
  <div
    className="flex-grow"
  />
  <div>
    <button
      className="button button-primary button-large action-button"
      onClick={[Function]}
    >
      Update
    </button>
  </div>
</div>
`;
